includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -718,8 +718,30 @@ public function maybe_process_upe_redirect() {
 				if ( isset( $_GET['redirect_status'] ) && 'succeeded' === $_GET['redirect_status'] ) {
 					$customer = new WC_Stripe_Customer( wp_get_current_user()->ID );
 					$customer->clear_cache();
-
 					wc_add_notice( __( 'Payment method successfully added.', 'woocommerce-gateway-stripe' ) );
+
+					// The newly created payment method does not inherit the customers' billing info, so we manually
+					// trigger an update; in case of failure we log the error and continue because the payment method's
+					// billing info will be updated when the customer makes a purchase anyway.
+					try {
+						$setup_intent_id = isset( $_GET['setup_intent'] ) ? wc_clean( wp_unslash( $_GET['setup_intent'] ) ) : '';
+						$setup_intent = WC_Stripe_API::request( [], 'setup_intents/' . $setup_intent_id, 'GET' );
+
+						$customer_data = WC_Stripe_Customer::map_customer_data( null, new WC_Customer( wp_get_current_user()->ID ) );
```

Comment:
```suggestion
						$customer_data = WC_Stripe_Customer::map_customer_data( null, $customer );
```
Can we not reuse the `$customer` instance that gets created above before the `try` block, instead of creating a new one here?

Reply:
We can't use `$customer` because `map_customer_data(...)` expects a `WC_Customer` and not a `WC_Stripe_Customer` as the second param.

Despite what the name may suggest, `WC_Stripe_Customer` does not extend `WC_Customer` 😞

Reply:
Whoops, you're totally right, I misread that! Ignore my suggestion in that case.